pkg/dataplane/itemscursor.go

```diff
@@ -32,15 +40,127 @@ type ItemsCursor struct {
 	getItemsInput   *GetItemsInput
 	container       Container
 	scattered       bool
+
+	logger        logger.Logger
+	retryAttempts int
+	retryInterval time.Duration
+}
+
+func getFunctionName(fn interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+}
+
+func retryFuncWithResult(ctx context.Context,
+	loggerInstance logger.Logger,
+	attempts int,
+	retryInterval time.Duration,
+	fn func(attempt int) (interface{}, bool, error)) (interface{}, error) {
```

Comment:
just to understand - the fn "attempt" parameter is just for future use and is not used currently, right?